src/actions/AccountReferralActions.ts

```diff
@@ -30,8 +31,12 @@ export function loadAccountReferral(account: EdgeAccount): ThunkAction<Promise<v
         account.disklet.getText(ACCOUNT_REFERRAL_FILE)
       ])
       const cache = asDiskReferralCache(JSON.parse(cacheText))
-      const referral = unpackAccountReferral(JSON.parse(referralText))
+      const referral = await unpackAccountReferral(JSON.parse(referralText))
```

Comment:
The `unpack` function should not be async - it's job is just to translate what we loaded earlier.

I think you should move the `getActivePromoIds` call out here, in the `loadAccountReferral` function. Then you can merge the active promo ID's with the return value from `unpackAccountReferral`.

Edit: Actually, it looks like `activePromotions` doesn't need to be a part of the account referral anyhow, so you don't need to merge the `getActivePromoIds` and `unpackAccountReferral` return values - they can just live next to each other.